Source/Mod/Editor/Definition/EditorDefinition.cs

```diff
@@ -0,0 +1,53 @@
+using ImGuiNET;
+
+namespace Celeste64.Mod.Editor;
+
+public abstract class EditorDefinitionData
+{
+	// Used to link back to the EditorDefinition
+	// TODO: Maybe remove this??
+	public string DefinitionFullName { get; set; }
+	
+	// TODO: Figure out how to let definitions mark support for certain special properties like position / rotation / scale
+	public virtual Vec3 Position { get; set; } = Vector3.Zero;
+	public virtual Vec3 Rotation { get; set; } = Vector3.Zero;
+	public virtual Vec3 Scale { get; set; } = Vector3.One;
+}
+
+public abstract class EditorDefinition
+{
```

Comment:
Is the idea that every actor would need its own Editor definition set up for it? Or is there going to be a common definition that is able to load all actors?
It would be nice if we could reuse as much of the actor setup as possible from the existing structure, so modders don't have to define all their actor data twice if they want to be able to use it in both formats, like for common actors shared between multiple mods. Or make it easy to support both with just a few attributes or something.

Reply:
Also, just a minor thought. If we do expect to need to create a lot of editor definitions for things, we may want to come up with a more clear name for it, because it wasn't immediately clear to me what this was. The editor prefix made me think it's a component of the editor itself. Maybe something like ActorDefinition might be more clear? (if that's not too confusing with the base actor class)